### PR TITLE
Flock sentinel modifications

### PR DESCRIPTION
**Pull request by FlameArrow57**
[BUG][CLEANLINESS][PERFORMANCE]

## About the PR 
This PR makes changes to the Flock sentinel structure, its code, and fixes a bug.

Makes cleanliness changes to the code.

Allows sentinels to target any flock enemy rather than just mobs.

Makes it so that sentinels always show their rays, depending on charge level, rather than only when they have just attacked.

sentinel

Adds another charge status message for when losing charge.

Adds in some checks for performance.

Fixes a bug where sentinels weren't losing charge when they were supposed to.

## Why's this needed? 
Code and performance improvements

Thought it would be cool for them to target vehicles, for rays to show when charged, and another status message would be nice.

Fixes a bug

### CODE REVIEW

File: code/obj/flock/structure/sentinel.dm [hunk 1]
```diff
@@ -42,93 +40,110 @@
...
+						break
+				if (!found_chain_target)
+					break
+			src.charge = 0
+			src.charge_status = CHARGING
 	else
-		if(charge > 0)//if theres charge make it decrease with time
-			src.charge_status = LOSING_CHARGE
+		if(src.charge > 0)
 			src.charge(-5)
```
Comment (TobleroneSwordfish) [inline code/obj/flock/structure/sentinel.dm:122]:
```suggestion
			src.charge(-5 * mult)
```

**Review by TobleroneSwordfish — APPROVED** [review]
lgtm, I'm glad we've banished `((-(cos(180*(3/100))-1)/2)*32)` from the code

**Review by frawhst — COMMENTED** [review]
good changes overall

File: code/obj/flock/structure/sentinel.dm [hunk 2]
```diff
@@ -42,93 +40,110 @@
...
-		if(charge > 0)//if theres charge make it decrease with time
+		if(src.charge > 0)
+			src.charge(-5 * mult)
 			src.charge_status = LOSING_CHARGE
-			src.charge(-5)
-		else
-			if(icon_state != "sentinel") icon_state = "sentinel"//forgive me again
-			src.charge_status = NOT_CHARGED //out of juice its dead
+		if (src.charge == 0)
+			if(icon_state != "sentinel")
```
[the remark this answers is not in the document]
Comment (FlameArrow57) [reply code/obj/flock/structure/sentinel.dm]: It was for the first time transition I think from sentinelon to sentinel

File: code/obj/flock/structure/sentinel.dm [hunk 3]
```diff
@@ -42,93 +40,110 @@
...
+	src.charge = clamp(chargeamount + charge, 0, 100)
 
 /obj/flock_structure/sentinel/proc/updatefilter()
-	var/filter = rays.get_filter("flock_sentinel_rays")
-	if(charge > 2)//else it just makes the sprite invisible, due to floats. this is small enough that it doesnt even showup anyway since its under the sprite
-		animate(filter, size=((-(cos(180*(charge/100))-1)/2)*32), time=10 SECONDS, flags = ANIMATION_PARALLEL)
-	else
-		animate(filter, size=((-(cos(180*(3/100))-1)/2)*32), time=10 SECONDS, flags = ANIMATION_PARALLEL)
+	var/filter = src.rays.get_filter("flock_sentinel_rays")
+	animate(filter, size = ((-(cos(180 * (charge / 100)) - 1) / 2) * 32), flags = ANIMATION_PARALLEL)
```
[the remark this answers is not in the document]
Comment (FlameArrow57) [reply code/obj/flock/structure/sentinel.dm:137]: Added if looks okay

[the remark this answers is not in the document]
Comment (frawhst) [reply code/obj/flock/structure/sentinel.dm:135]: lgtm

**Comment by frawhst** [comment]
oh update on the filter thing: you can cast a filter as `var/dm_filter` to access the vars rather than using `:`. That type is nowhere in the ref but coderbus informed me.

**Comment by frawhst** [comment]
so obscure the linter doesn't even know about it. nice

**Comment by frawhst** [comment]
PR is good, I'll merge this as soon as a new SpacemanDMM version is released (maintainer is aware of the issue) so the linter isn't screaming constantly. I'll resolve merge conflicts and whatever since my request is causing the holdup.

**Comment by frawhst** [comment]
`UNLINT` my beloved
IDB2 is sleeping and there's a secret ref in there but we'll live with it

**Pull request merged**